### DISCUSSION TITLE
CubeFS performance (so much potential!)

### DISCUSSION THREAD

**Discussion by Zorlin** [cubefs/cubefs#2830] [General]
Hi there,

Long time user of many other DFS here.

I spun up a CubeFS installation out of curiosity.

It consists of the following data nodes:
* 201 - 13 disks
* 202 - 12 disks
* 203 - 13 disks
* 204 - 12 disks
* 207 - 26 disks

Each node is configured with the latest CubeFS, dual 24-core EPYC 7443 processors, at least 512GiB of RAM, and Micron 7450 NVMe U.3 SSDs for the metadata storage.

Each node is on dual 25gbps networking, although currently only one link is active on each (active-backup policy). This is all in one rack, direct attach copper with high quality components.

What we see is that when given a *very* short read run, we see performance faster than 1.3 GiB/s - a very good result.

```
Run status group 0 (all jobs):
   READ: bw=1346MiB/s (1411MB/s), 30.9MiB/s-144MiB/s (32.4MB/s-151MB/s), io=1856MiB (1946MB), run=1001-1379msec
```

However, when given a longer run, we see a much worse result:
```
Run status group 0 (all jobs):
   READ: bw=234MiB/s (245MB/s), 7919KiB/s-25.5MiB/s (8109kB/s-26.8MB/s), io=7212MiB (7562MB), run=30128-30813msec
```

With speeds dropping as low as 30MiB/s on a 10 minute run.

The benchmark I used - fio - was this command - `fio --name=seqread --rw=read --bs=4M --numjobs=16 --size=2G --runtime=30s --time_based` - with a normal 3-replica volume.

Any advice on how we can improve this is appreciated - if I can get sustained high speeds out of CubeFS I would like to start using it more!

**Comment by leonrayang** [comment]
Hello @Zorlin , sorry for the late reply. This test result is indeed abnormal. Generally speaking, reading will not decline. For convenience, you can join cubefs’s slack channel（cubefs.slack.com). Let’s take a look at your log. or email me directly,thank you.[changliang@oppo.com](mailto:changliang@oppo.com)

**Reply by Zorlin** [reply]
Hi, sorry, I no longer have access to the cluster I tried this on. I'm using CubeFS as my main production home storage system soon though, which I'm nervous but excited about :+1:
